perun-rpc/src/main/java/cz/metacentrum/perun/rpc/methods/GroupsManagerMethod.java

```diff
@@ -314,21 +316,108 @@ public Void call(ApiCaller ac, Deserializer parms) throws PerunException {
+			}
+			Group group = ac.getGroupById(parms.readInt("group"));
+			List<Member> members = new ArrayList<>();
+			for (Integer memberInt : memberInts) {
+				try {
+					Member member = ac.getGroupsManager().getGroupMemberById(ac.getSession(), group, memberInt);
+					if (MembershipType.DIRECT.equals(member.getMembershipType())) {
+						members.add(member);
+					}
+				} catch(NotGroupMemberException e) {
+					//just skipped
+				}
+			}
+			ac.getGroupsManager().removeMembers(ac.getSession(),
+				group,
+				members);
+			return null;
+		}
+	},
+
+
 	/*#
 	 * Removes a member from a group.
 	 *
+	 * @throw MemberNotExistsException When member doesn't exist
+	 * @throw NotGroupMemberException  When member is not in the group
+	 * @throw GroupNotExistsException When the group doesn't exist
+	 * @throw WrongAttributeAssignmentException When assigning atribute to wrong entity
+	 * @throw AttributeNotExistsException When attribute doesn't exist
+	 * @throw ExternallyManagedException When the group is externally managed
+	 *
 	 * @param group int Group <code>id</code>
 	 * @param member int Member <code>id</code>
 	 */
+	/*#
+	 * Removes a member from groups. If a member is not in the group or is indirect, it is skipped, but the rest of groups are processed.
+	 * Non-empty list of groups expected. In case of empty list, member is not removed.
+	 *
+	 * @throw MemberNotExistsException When member doesn't exist
+	 * @throw NotGroupMemberException  When member is not in the group
+	 * @throw GroupNotExistsException When the group doesn't exist
+	 * @throw WrongAttributeAssignmentException When assigning atribute to wrong entity
+	 * @throw AttributeNotExistsException When attribute doesn't exist
+	 * @throw ExternallyManagedException When the group is externally managed
+	 *
+	 * @param groups List<Integer> Group - can be empty <code>id</code>
+	 * @param member int Member <code>id</code>
+	 */
 	removeMember {
-
 		@Override
 		public Void call(ApiCaller ac, Deserializer parms) throws PerunException {
 			ac.stateChangingCheck();
-
-			ac.getGroupsManager().removeMember(ac.getSession(),
+			if (parms.contains("group")) {
+				ac.getGroupsManager().removeMember(ac.getSession(),
 					ac.getGroupById(parms.readInt("group")),
 					ac.getMemberById(parms.readInt("member")));
+			} else if (parms.contains("groups")) {
+				List<Integer> groupsInts = parms.readList("groups", Integer.class);
+				if (groupsInts == null) {
+					throw new RpcException(RpcException.Type.MISSING_VALUE, "Non-empty list of groups not sent.");
+				}
+				List<Group> groups = new ArrayList<>();
+				for (Integer groupInt : groupsInts) {
+					try {
+						Group group = ac.getGroupById(groupInt);
+						Member member = ac.getGroupsManager().getGroupMemberById(ac.getSession(), group, parms.readInt("member"));
+						if (MembershipType.DIRECT.equals(member.getMembershipType())) {
+							groups.add(group);
+						}
+					} catch(NotGroupMemberException e) {
+						//just skipped
```

Comment:
Please, write also the reason for skipping there "skipped because user is not member of this group so we don't need to remove him".

Reply:
done